app/main.py

```diff
@@ -1,3 +1,73 @@
+from __future__ import annotations
+from typing import Optional
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: "Distance" | int | float) -> Optional["Distance"]:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+
+        if isinstance(other, (int, float)):
+            return Distance(self.km + other)
+
+    def __iadd__(
+            self,
+            other: "Distance" | int | float
+    ) -> Optional["Distance"]:
+        if isinstance(other, Distance):
+            self.km += other.km
+
+        if isinstance(other, (int, float)):
+            self.km += other
+
+        return self
+
+    def __mul__(self, other: int) -> Optional["Distance"]:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: int | float) -> Optional["Distance"]:
+        return Distance(round(self.km / other, 2))
```

Comment:
Maybe here we can make annotation of function as -> Distance, because I don't see the way how we can get None return.

Reply:
Noted! )